src/com/company/Cart.java

```diff
@@ -0,0 +1,60 @@
+package com.company;
+
+import java.util.List;
+
+public class Cart {
+    private List<Product> products;
+    private Client client;
+
+    public Cart(List<Product> products, Client client) {
+        this.products = products;
+        this.client = client;
+    }
+
+    public Cart(Client client) {
+        this.client = client;
+    }
+
+    public List<Product> getProducts() {
+        return products;
+    }
+
+    public void setProducts(List<Product> products) {
+        this.products = products;
+    }
+
+    public Client getClient() {
+        return client;
+    }
+
+    public void setClient(Client client) {
+        this.client = client;
+    }
+
+    public void addToCart(Product product) {
+        if (product.getStock() > 0) {
+            this.products.add(product);
+            product.decreaseStock(1);
+        } else {
+            throw new IllegalArgumentException("not enough in stock");
```

Comment:
Taki message w exceptionie troche malo mowi, poprawniejszym podjesciem bylloby napisanie pelniejszej informacji typu: "cant add - product with id {id} has not enough..." - chodzi o to zeby dalo sie zidentyfikowac problem biznesowo